specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,389 @@
+WinRT API in the Reunion AppLifecycle component, and include additional low-level notifications
+based on the Win32
+[PowerSettingRegisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingregisternotification)
+and matching
+[PowerSettingUnregisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingunregisternotification)
+APIs.
+
+<br>
+
+## Description
+
+Providing a Reunion version of PowerManager is relatively straightforward, in that we aim to bring
+across the entire existing API in its current form. For the Win32 APIs, the situation is a little
+murkier. It boils down to a choice between 2 usage patterns:
+
+-   [RegisterPowerSettingNotification](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerpowersettingnotification)
+    uses a Windows message-based model. When the app registers for notifications, the system will
+    send out WM_POWERBROADCAST messages.
+-   [PowerSettingRegisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingregisternotification)
+    uses a callback model. When the app registers for notifications, it supplies a pointer to a
+    callback function, and the system will call back to this function instead of sending messages.
+
+Classic Win32 apps can use either the WM model and/or the callback model. The callback model is more
+consistent with other app types - including not only modern UWP apps, but also traditional MFC,
+Winforms and WPF apps. Also, the callback model does not prescribe a windowing or threading model,
+which is better for headless/background components. Given this, since all app types can work with
+the callback model, but only classic Win32 apps use the WM model, the Reunion API will use the
+callback model for Cobalt. Additional WM-based behavior can be added in a later release, if customer
+demand indicates it would provide significant benefit.
+
+The net is that the Reunion PowerManager API will be a combination of WinRT PowerManager and Win32
+PowerSettingRegisterNotification. We will define new WinRT types (in
+Microsoft.ApplicationModel.PowerManager) that provide event handlers for apps to use.
+
+## Examples
+
+### Subscribing and responding to events
+
+The power notifications are relatively simple. The app provides a callback function for each
+notification it wants to receive, and when the event is raised the system invokes that callback. The
+app could start listening for the notifications at any time; it would most often do this early in
+its life, perhaps in WinMain - but the AppLifecycle mechanism does not impose any constraints on
+this, and the app is free to subscribe or unsubscribe at any time.
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
```

Comment:
Do we need to CoInitialize? Or can these methods operate without COM?

Reply:
RECOMMEND: See previous answers

Reply:
Fixed.